app/models/protocol.rb

```diff
@@ -185,30 +187,46 @@ def unique_rm_id_to_protocol
     ]
   )
 
-  scope :search_query, -> (search_term) {
+  scope :search_query, lambda { |search_attrs|
     # Searches protocols based on short_title, title, id, and associated_users
     # Protects against SQL Injection with ActiveRecord::Base::sanitize
 
     # inserts ! so that we can escape special characters
-    escaped_search_term = search_term.to_s.gsub(/[!%_]/) { |x| '!' + x }
+    escaped_search_term = search_attrs[:search_text].to_s.gsub(/[!%_]/) { |x| '!' + x }
 
     like_search_term = ActiveRecord::Base::sanitize("%#{escaped_search_term}%")
-    exact_search_term = ActiveRecord::Base::sanitize(search_term)
-
-    #TODO temporary replacement for "MATCH(identities.first_name, identities.last_name) AGAINST (#{exact_search_term})"
-    where_clause = ["CONCAT(identities.first_name, ' ', identities.last_name) LIKE #{like_search_term} escape '!'"]
-
-    where_clause += ["protocols.short_title like #{like_search_term} escape '!'",
-      "protocols.title like #{like_search_term} escape '!'",
-      "protocols.id = #{exact_search_term}"]
-
-    where_clause += ["human_subjects_info.hr_number = #{exact_search_term}", 
-                      "human_subjects_info.pro_number = #{exact_search_term}"]
-
-    joins(:identities).joins(:human_subjects_info).
+    exact_search_term = ActiveRecord::Base::sanitize(search_attrs[:search_text])
+
+    case search_attrs[:search_drop]
+      
+    when "PI"
+      where_clause = ["CONCAT(identities.first_name, ' ', identities.last_name) LIKE #{like_search_term} escape '!'"]
+      joins(:principal_investigators).
+        where(where_clause.compact.join(' OR ')).
```

Comment:
A lot of these `where_clause`'s only have one element. So invoking `.compact.join(' OR ')` has the same effect as `where_clause[0]`. Maybe you can simplify things a lot by just defining some of these `where_clause`'s as a string.